Missing build_export dependencies?

Hi there,

we recently experienced build errors in our package where we build moveit in the upstream workspace in CI:

```
--- stderr: moveit_ros_trajectory_cache
  moveit_ros_trajectory_cache: You did not request a specific build type: Choosing 'Release' for maximum performance
  CMake Error at /opt/ros/rolling/share/warehouse_ros_sqlite/cmake/ament_cmake_export_dependencies-extras.cmake:21 (find_package):
    By not providing "Findsqlite3_vendor.cmake" in CMAKE_MODULE_PATH this
    project has asked CMake to find a package configuration file provided by
    "sqlite3_vendor", but CMake did not find one.
  
    Could not find a package configuration file provided by "sqlite3_vendor"
    with any of the following names:
  
      sqlite3_vendorConfig.cmake
      sqlite3_vendor-config.cmake
  
    Add the installation prefix of "sqlite3_vendor" to CMAKE_PREFIX_PATH or set
    "sqlite3_vendor_DIR" to a directory containing one of the above files.  If
    "sqlite3_vendor" provides a separate development package or SDK, be sure it
    has been installed.
  Call Stack (most recent call first):
    /opt/ros/rolling/share/warehouse_ros_sqlite/cmake/warehouse_ros_sqliteConfig.cmake:41 (include)
    CMakeLists.txt:59 (find_package)
```

My interpretation is that since `warehouse_ros_sqlite` exports the dependency to `sqlite3_vendor` in the [`CMakeLists.txt`](https://github.com/moveit/warehouse_ros_sqlite/blob/3326e668dfb9172d89f8406918b3564aea1a9b74/CMakeLists.txt#L93) but not in the `package.xml` the package is indeed missing.

I should state that we don't build `warehouse_ros_sqlite` in our workspace, but only `moveit2` and `moveit_msgs`. However, this doesn't seem like a problem that we are building/linking against not the latest `warehouse_ros_sqlite` version.

We worked around this issue by building `warehouse_ros_sqlite` from source in the CI, as well, but this feels like something that could be improved nevertheless. Either by adding `build_export_depend` entries to the package.xml or by removing them from the `ament_export_dependencies` call. I haven't checked the code deeply enough to know which one would be preferrable.

--- a/package.xml
+++ b/package.xml
@@ -12,15 +12,12 @@
 
   <buildtool_depend>ament_cmake</buildtool_depend>
 
-  <build_depend>sqlite3_vendor</build_depend>
   <build_depend>boost</build_depend>
-  <build_depend>libsqlite3-dev</build_depend>
 
   <depend>warehouse_ros</depend>
   <depend>class_loader</depend>
   <depend>rclcpp</depend>
-
-  <exec_depend>sqlite3</exec_depend>
+  <depend>sqlite3_vendor</depend>
 
   <test_depend>ament_cmake_copyright</test_depend>
   <test_depend>ament_cmake_gtest</test_depend>

